torch/csrc/dynamo/eval_frame.c

```diff
@@ -580,6 +585,13 @@ static PyObject* _custom_eval_frame(
     DEBUG_TRACE("skip %s", get_frame_name(frame));
     return eval_frame_default(tstate, frame, throw_flag);
   }
+  if (extra == SKIP_CODE_RECURSIVE) {
+    DEBUG_TRACE("skip recursive %s", get_frame_name(frame));
+    skip_code_recursive = true;
```

Comment:
Why do we need a special flag for this?  Can't we just disable dynamo temporarily?

Reply:
If we use `eval_frame_callback_set(Py_None);`, then any nested calls with a @compile decorator will re-enable tracing. Having a separate flag forces the eval frame callback to continue to be disabled in this case.

Reply:
We will match the skip code recursive semantics with the disable decorator for now (will update when the disable decorator semantics change).